lib/VeresOneClient.js

```diff
@@ -1,130 +1,87 @@
-   *   DID Document from cryptonym.
+   * @param {object} options - Options hashmap.
+   * @param {string} options.did - DID authority uri (without hash fragment).
    *
    * @returns {Promise<object>} Resolves to DID Document Fetch Result
    */
-  async get({did, forceConstruct = false}) {
+  async get({did} = {}) {
     if(!did) {
-      throw new Error('Invalid or missing DID URI.');
-    }
-    const [docUri, hashFragment] = did.split('#');
-    const isNym = (did.startsWith('did:v1:nym:') ||
-      did.startsWith('did:v1:test:nym:'));
-    let result = {};
-    let didDoc;
-
-    // FIXME - remove, replace with cache
-    if(isNym && forceConstruct) {
-      didDoc = (await VeresOneDidDoc.fromNym({did: docUri})).doc;
-    } else {
-      try {
-        result = await this.ledger.getRecord({id: docUri});
-        didDoc = result.record;
-      } catch(error) {
-        if(error.name === 'NotFoundError' && isNym) {
-          didDoc = (await VeresOneDidDoc.fromNym({did: docUri})).doc;
-        } else {
-          throw error;
-        }
-      }
+      throw new TypeError('Invalid or missing DID URI.');
     }
+    const {record: didDocument} = await this.ledger.getRecord({id: did});
 
-    const context = didDoc['@context'];
-
-    if(!hashFragment) {
-      // full DID Doc
-      result.type = 'LedgerDidDocument';
-      result.doc = didDoc;
-    } else {
-      // Request for a subgraph (likely just the key node)
-      const map = await jsonld.createNodeMap(didDoc);
-      const subGraph = map[did];
-      if(!subGraph) {
-        const error = new Error(
-          `Failed to get subgraph within a DID Document, uri: "${did}".`
-        );
-        error.name = 'NotFoundError';
-        throw error;
-      }
-
-      // result.type = 'Key'; <- not sure what this should be
-      result.doc = await jsonld.compact(subGraph, context);
-    }
-
-    return result;
+    return didDocument;
   }
 
+  /**
+   * Get the status of a Document from the ledger Agent Status Service.
+   *
+   * @returns {Promise<object>} A document with a status.
+   */
   async getStatus() {
     return this.ledger.getStatus();
   }
 
   async getTicketServiceProof({operation, ticketService}) {
-    const baseURL = ticketService;
-    const ticketServiceApi = create({baseURL});
-    const response = await ticketServiceApi.post(
-      '/', {operation}, {httpsAgent: this.httpsAgent});
-    if(response.problem) {
-      const error = new VeresOneClientError(
-        'Error retrieving record.', 'NetworkError');
-      if(response.problem === 'CLIENT_ERROR') {
-        error.details = {
-          baseURL, error: response.data, status: response.status
-        };
-      } else {
-        error.details = {
-          baseURL, error: response.originalError, status: response.status
-        };
-      }
+    let result;
+    try {
+      result = await httpClient.post(ticketService, {
+        json: operation, httpsAgent: this.httpsAgent
```

Comment:
This is corrected in my PR, but ticketService expects the post body to be in the form of:

```js
{
  "operation": {
    "@context": "foo"
  }
}
```

see the original code in master: https://github.com/veres-one/did-veres-one/blob/798fea3473b3b2c5dfaeedceaf10324accfe8c5b/lib/VeresOneClient.js#L110-L113